python/tank_vendor/shotgun_deploy/configuration.py

```diff
@@ -237,15 +238,36 @@ def get_tk_instance(self, sg_user):
 
         :param sg_user: Authenticated Shotgun user to associate
                         the tk instance with.
+
         """
-        # @todo - check if there is already a tank instance, in that case unload or warn
-        path = self._paths[sys.platform]
-        core_path = os.path.join(path, "install", "core", "python")
-        sys.path.insert(0, core_path)
-        import tank
+
+        core_path = os.path.join(self.path, "install", "core", "python")
+
+        # if the custom import handler has been added to `sys.meta_path`, then
+        # the import should work with the proper version of core. if not, then
+        # fallback to the behavior of manually adding this configuration's
+        # python dir to `sys.path` in order to pick it up.
+        try:
+            # custom import handler will handle this properly
+            import tank
+        except ImportError:
+            # oops, import handler likely not installed. fall back to using
```

Comment:
The import handler, once installed, acts on any importable packages/modules it finds in the directory provided. So when I tell it to use `/path/to/my/core`, it sees what in that directory is importable (via a call to `imp.find_module` with the files/directories in there). If python thinks they're importable, then the import handler knows to attempt imports in those namespaces. So if some code calls `import tank_vendor`, then the handler will process it. If the code calls `import maya.cmds`, then it will ignore it and allow the default logic to handle that. Does that help?